File: 169.majority-element.py
#
# @lc app=leetcode id=169 lang=python3
#
# [169] Majority Element
#

# @lc code=start
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash = {}
        for num in nums:
            if num not in hash: hash[num] = 0
            hash[num] += 1
        return sorted(hash.items(), key=lambda kv: kv[1])[-1][0]
# @lc code=end

File: 2020/169.majority-element.js
/*
 * @lc app=leetcode id=169 lang=javascript
 *
 * [169] Majority Element
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
  let freq = {}
  for (let i=0;i<nums.length;i++) {
    let v = nums[i]
    freq[v] = (freq[v]|0) + 1
    if (freq[v] > nums.length/2) return v
  }
  return false
};
// @lc code=end

